com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/ServiceCreator.java

```diff
@@ -33,14 +35,15 @@ public void createService(CloudFoundryOperations client, CloudServiceExtended se
         LOGGER.debug(format("Service \"{0}\" has defined service offering alternatives \"{1}\" for default service offering \"{2}\"",
             service.getName(), service.getServiceAlternatives(), service.getLabel()));
         List<String> possibleServiceOfferings = computePossibleServiceOfferings(service);
-        List<String> existingServiceOfferings = client.getServiceOfferings().stream().map(
-            cloudServiceOfferingExtended -> cloudServiceOfferingExtended.getLabel()).collect(Collectors.toList());
-        List<String> validServiceOfferings = computeValidServiceOfferings(client, possibleServiceOfferings, existingServiceOfferings);
+        Map<String, List<CloudServicePlan>> existingServiceOfferings = client.getServiceOfferings().stream().collect(
+            Collectors.toMap(CloudServiceOffering::getName, CloudServiceOffering::getCloudServicePlans));
+        List<String> validServiceOfferings = computeValidServiceOfferings(client, possibleServiceOfferings, service.getPlan(),
+            existingServiceOfferings);
 
         if (CollectionUtils.isEmpty(validServiceOfferings)) {
             LOGGER.error(format(
-                "Service \"{0}\" could not be created because any of defined service offering(s) \"{1}\" does not match with existing service offerings \"{2}\"",
-                service.getName(), possibleServiceOfferings, existingServiceOfferings));
+                "Service \"{0}\" could not be created because none of the service offering(s) \"{1}\" match with existing service offerings \"{2}\" or provide service plan \"{3}\"",
```

Comment:
This won't be shown to the user, so what will he see if his service cannot be created? Nothing?

Reply:
Done.
Despite of the logged error, the exception will have more meaningful exception message.